src/System.Private.CoreLib/shared/System/Collections/Generic/ArraySortHelper.cs

```diff
@@ -49,7 +49,7 @@ internal partial class ArraySortHelper<T>
     {
         #region IArraySortHelper<T> Members
 
-        public void Sort(T[] keys, int index, int length, IComparer<T>? comparer)
+        public void Sort(Span<T> keys, int index, int length, IComparer<T>? comparer)
         {
             Debug.Assert(keys != null, "Check the arguments in the caller!");
```

Comment:
Comparing Span with null does not make sense (always true).

Reply:
(multiple places)

Reply:
Comparing Span with null does actually work albeit with overhead. I agree it doesn't make sense on this specific place.

If you convert a `null` array to Span and then compare the `span == null` it will return true. The RHS will implicitly get converted to Span and the comparison of the two spans will yield true if the LHS was also created from `null` array.

Reply:
I understand that. Just wanted to point out that it's not a no-op as suggested.